File: Core/Entities/Product.cs
using Application.Entities;
using Core.HelperTypes;
using System.ComponentModel.DataAnnotations;

namespace Core.Entities;

public class Product : FullAuditableEntity, IPrice
{
    [Required]
    public string Name { get; set; }

    [Required]
    public string Description { get; set; }

    [Required]
    public decimal Price { get; set; }
    public Category Category { get; set; }
    public int CategoryId { get; set; }
    public ProductBrand ProductBrand { get; set; }
    public int ProductBrandId { get; set; }
    public AppUser User { get; set; }
    public int UserId { get; set; }
    public County County { get; set; }
    public int CountyId { get; set; }
    public bool Showcase { get; set; }
    public bool IsActive { get; set; }
    public CurrencyCode Currency { get; set; }
    public ProductMachine ProductMachine { get; set; }
    public ICollection<Photo> Photos { get; set; }
}

File: client/src/app/shared/models/product.ts
import { ICategory } from './category';
import { Member } from './member';
import { Photo } from './photo';

export interface IProduct {
  id: number;
  name: string;
  description: string;
  price: number;
  priceText: string;
  pictureUrl: string;
  productType: string;
  created: Date;
  city: string;
  county: string;
  isActive: boolean;
  // category: ICategory;
  categoryId: number;
  photos: Photo[];
  user: Member;
  isFavourite: boolean;
}
